File: src/main/java/de/maggiwuerze/xdccloader/persistency/UserRepository.java
package de.maggiwuerze.xdccloader.persistency;

import de.maggiwuerze.xdccloader.model.entity.User;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;


@Repository
public interface UserRepository extends PagingAndSortingRepository<User, Long> {

    List<User> findAll();
    Optional<User> findUserByName(String name);

}

File: src/main/kotlin/de/maggiwuerze/xdccloader/persistency/UserRepository.kt
package de.maggiwuerze.xdccloader.persistency

import de.maggiwuerze.xdccloader.model.User
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository
import java.util.*


@Repository
interface UserRepository : PagingAndSortingRepository<User, Long> {


    fun findUserByName(name : String) : Optional<User>

}